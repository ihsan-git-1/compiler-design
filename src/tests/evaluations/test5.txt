class test {
   int x=2;
   double y=3.5;

   double z = x+y;
   double w = x-y;
   double r = x*y;
   double t = x/y;


 }