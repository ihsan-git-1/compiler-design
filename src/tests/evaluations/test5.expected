
*********** AST ************

Line : (1) Top Tree Declaration  Child Count =  1 Type = Top Tree
Line : (1) Class Declaration Name is test parent AllClassesDeclaration Child Count =  10 Type = class
Line : (2) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (2) Integer Declaration Line: Number of declarations on this line: 1 Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type integer)
 Line : (3) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (3) Double Declaration Line: Number of declarations on this line: 1Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type double)
 Line : (5) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (5) Double Declaration Line: Number of declarations on this line: 1Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type double)
 Line : (6) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (6) Double Declaration Line: Number of declarations on this line: 1Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type double)
 Line : (7) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (7) Double Declaration Line: Number of declarations on this line: 1Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type double)
 Line : (8) Dart Variable Declaration:  parent: ClassDeclaration Child Count =  1 Type = object
Line : (8) Double Declaration Line: Number of declarations on this line: 1Number of assignments on this line: 1 parent: Variable Child Count =  2 Type = variable declaration line (type double) ,  

*********** SYMBOL TABLE ************

Global Scope (0)
Class Scope (1) Parent Is Global Scope (0)

Identifier x, Type int, Value : 2 , Scope Class Scope (1)
Identifier y, Type double, Value : 3.5 , Scope Class Scope (1)
Identifier z, Type double, Value : 5.5 , Scope Class Scope (1)
Identifier w, Type double, Value : -1.5 , Scope Class Scope (1)
Identifier r, Type double, Value : 7.0 , Scope Class Scope (1)
Identifier t, Type double, Value : 0.5714285714285714 , Scope Class Scope (1)
